### PR TITLE
Feature/hook up signing

### PR DESCRIPTION
**Pull request by frapperino**

### CODE REVIEW

File: src/pages/Sign/sections/SignUp.tsx [hunk 1]
```diff
@@ -137,62 +196,216 @@ export const SignUp: React.SFC = () => (
...
+                                  <div>
+                                    <ErrorText>
+                                      <TranslationsConsumer textKey="SIGN_BANKID_STANDARD_ERROR_MESSAGE">
+                                        {(errorText) => errorText}
+                                      </TranslationsConsumer>
+                                    </ErrorText>
+                                  </div>
+                                )
+                              }
+                              if (data) {
```
Comment (palmfjord) [inline src/pages/Sign/sections/SignUp.tsx]: can we move these nested branching statements into a util functions?
a) nesting branching statements gets quite hard to read
b) We're mixing view and business logic here imo
c) I don't like branching in general, so if we could solve this in another approach i would be very happy :D

File: src/pages/Sign/sections/SignUp.tsx [hunk 2]
```diff
@@ -137,62 +196,216 @@ export const SignUp: React.SFC = () => (
...
+                                      }
+                                    } else {
+                                      return null
+                                    }
+                                  case SIGNSTATE.COMPLETED: // TODO: backend sets this to null atm
+                                    if (
+                                      dataStatus.status ===
+                                      BANKIDSTATUS.COMPLETE
+                                    ) {
+                                      window.location.href = '/download'
```
Comment (palmfjord) [inline src/pages/Sign/sections/SignUp.tsx]: this should probably be a `Redirect` component from `react-router-dom` so we don't loose what we have in memory

File: src/pages/Sign/sections/SignUp.tsx [hunk 3]
```diff
@@ -137,62 +196,216 @@ export const SignUp: React.SFC = () => (
...
+                                    } else {
+                                      return null
+                                    }
+                                  case SIGNSTATE.FAILED:
+                                    if (
+                                      dataStatus.status === BANKIDSTATUS.FAILED
+                                    ) {
+                                      switch (dataStatus.code) {
+                                        case 'expiredTransaction':
```
Comment (palmfjord) [inline src/pages/Sign/sections/SignUp.tsx]: why did we not make these into enums?

File: src/pages/Sign/sections/SignUp.tsx [hunk 4]
```diff
@@ -121,6 +131,55 @@ const UserSchema = Yup.object().shape({
     .required('SIGN_PERSONAL_NUMBER_REQUIRED'),
```
Comment (palmfjord) [inline src/pages/Sign/sections/SignUp.tsx]: Imo `UserSchema` -> `userSchema` as it is not static

File: src/pages/Sign/sections/SignUp.tsx (hunk 3 above)
Comment (frapperino) [reply src/pages/Sign/sections/SignUp.tsx]: They are strings in the BE so I wanted to be consistent

File: src/pages/Sign/sections/SignUp.tsx [hunk 5]
```diff
@@ -3,9 +3,13 @@ import {
   TranslationsConsumer,
   TranslationsPlaceholderConsumer,
 } from '@hedviginsurance/textkeyfy'
+import { SessionContainer } from 'containers/SessionContainer'
 import { Field, Form, Formik } from 'formik'
+import gql from 'graphql-tag'
 import * as React from 'react'
+import { Mutation, Subscription } from 'react-apollo'
 import styled from 'react-emotion'
+import { Redirect } from 'react-router'
```
Comment (palmfjord) [inline src/pages/Sign/sections/SignUp.tsx]: just fyi, this should be `react-router-dom` :) `react-router` works but only for legacy compatibility reasons and its deprecated

File: src/containers/OfferContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+export const OfferContainer: React.SFC<OfferContainerProps> = ({
+  children,
+}) => (
+  <Query<OfferData> query={OFFER_QUERY}>
+    {({ loading, error, data }) => {
+      if (loading) {
+        return <div>Loading</div>
+      }
+      if (error || !data) {
+        return <pre>{JSON.stringify(error, null, 2)}</pre>
```
Comment (hedvigoscar) [inline src/containers/OfferContainer.tsx]: We probably need to remove this 🤷‍♂️

File: src/containers/OfferContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+export const OfferContainer: React.SFC<OfferContainerProps> = ({
+  children,
+}) => (
+  <Query<OfferData> query={OFFER_QUERY}>
+    {({ loading, error, data }) => {
+      if (loading) {
+        return <div>Loading</div>
+      }
+      if (error || !data) {
```
Comment (hedvigoscar) [inline src/containers/OfferContainer.tsx]:
```suggestion
      if (error) {
```

File: src/containers/OfferContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+interface OfferContainerProps {
+  children: (offer: OfferData) => React.ReactNode
+}
+
+export const OfferContainer: React.SFC<OfferContainerProps> = ({
+  children,
+}) => (
+  <Query<OfferData> query={OFFER_QUERY}>
+    {({ loading, error, data }) => {
+      if (loading) {
```
Comment (hedvigoscar) [inline src/containers/OfferContainer.tsx]:
```suggestion
      if (loading || !data) {
```

File: src/pages/Sign/sections/SignSubscription.tsx [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+        case SIGNSTATE.IN_PROGRESS:
+          if (dataStatus.status === BANKIDSTATUS.PENDING) {
+            return <BankidStatus message={dataStatus.code} />
+          } else {
+            return null
+          }
+        case SIGNSTATE.COMPLETED:
+          if (dataStatus.status === BANKIDSTATUS.COMPLETE) {
+            return <Redirect to="/download" />
+          } else {
```
Comment (hedvigoscar) [inline src/pages/Sign/sections/SignSubscription.tsx]: Lets drop all these `else return null`-cases: if we have a `default: return null` we can rely on that case to improve conciseness and readability.

```suggestion

```

File: src/pages/Sign/sections/SignUp.tsx [hunk 10]
```diff
@@ -121,79 +149,137 @@ const UserSchema = Yup.object().shape({
...
+                      address: offer.insurance.address,
+                    }}
+                  >
+                    {(title) => title}
+                  </TranslationsPlaceholderConsumer>
+                )}
+              </OfferContainer>
+            </Header>
+          </HeaderWrapper>
+          <SessionContainer>
```
Comment (hedvigoscar) [inline src/pages/Sign/sections/SignUp.tsx]: We should remove this `<SessionContainer />` as it no longer adds any value, given our `<SessionTokenGuard />`at the top.

```suggestion

```

File: src/pages/Sign/sections/SignUp.tsx [hunk 11]
```diff
@@ -121,79 +149,137 @@ const UserSchema = Yup.object().shape({
...
+                              <TranslationsConsumer textKey="SIGN_INPUT_ONE_TITLE">
+                                {(title) => title}
+                              </TranslationsConsumer>
+                            </InputTitle>
+
+                            <InputField
+                              name="email"
+                              id="email"
+                              style={{
+                                borderBottom: touched.email
```
Comment (palmfjord) [inline src/pages/Sign/sections/SignUp.tsx]: this gets so hard to read... could we maybe move this out to a util function?

File: src/pages/Sign/sections/SignUp.tsx [hunk 12]
```diff
@@ -121,79 +149,137 @@ const UserSchema = Yup.object().shape({
...
+                            <InputTitle>
+                              <TranslationsConsumer textKey="SIGN_INPUT_TWO_TITLE">
+                                {(title) => title}
+                              </TranslationsConsumer>
+                            </InputTitle>
+                            <InputField
+                              name="personalNumber"
+                              id="personalNumber"
+                              style={{
+                                borderBottom: touched.personalNumber
```
Comment (palmfjord) [inline src/pages/Sign/sections/SignUp.tsx]: ...also we're kinda repeating ourselves. We should definitely move this out to a utils function

File: src/pages/Sign/sections/SignUp.tsx.orig [hunk 13]
```diff
@@ -0,0 +1,571 @@
+import { colors, fonts } from '@hedviginsurance/brand'
```
[the remark this answers is not in the document]
Comment (frapperino) [reply src/pages/Sign/sections/SignUp.tsx.orig]: omg

**Comment by hedvigoscar** [comment]
Lets get an internal launch rolling! :tada:

**Pull request merged**